src/core_plugins/elasticsearch/lib/set_headers.js

```diff
@@ -0,0 +1,15 @@
+import { isObject } from 'lodash';
+
+export default function setHeaders(originalHeaders, newHeaders) {
+  if (!isObject(originalHeaders)) {
```

Comment:
I was thinking more that you probably don't want either of these to be arrays, for example, right?